timeseries/tests/unittests/models/test_models.py

```diff
@@ -26,649 +23,479 @@
 
+    def test_when_val_score_accessed_then_value_is_set(self, trained_model):
+        assert isinstance(trained_model.val_score, float)
 
-@pytest.mark.parametrize("model_class", TESTABLE_MODELS)
-@pytest.mark.parametrize("prediction_length", [1, 5])
-def test_when_score_and_cache_oof_called_then_attributes_are_saved(model_class, prediction_length, trained_models):
-    model = trained_models[(prediction_length, repr(model_class))]
-    assert isinstance(model.val_score, float)
-    assert isinstance(model.predict_time, float)
+    def test_when_predict_time_accessed_then_value_is_set(self, trained_model):
+        assert isinstance(trained_model.predict_time, float)
 
+    def test_given_score_and_cache_oof_called_when_get_oof_predictions_called_then_oof_predictions_are_saved(
+        self,
+        trained_model,
+    ):
+        if isinstance(trained_model, MultiWindowBacktestingModel):
+            pytest.skip()
+
+        oof_predictions = trained_model.get_oof_predictions()[0]
+        assert isinstance(oof_predictions, TimeSeriesDataFrame)
+        oof_score = trained_model._score_with_predictions(DUMMY_TS_DATAFRAME, oof_predictions)
+        assert isinstance(oof_score, float)
+
+    def test_when_score_called_then_model_receives_truncated_data(self, trained_model):
+        with mock.patch.object(trained_model, "predict") as patch_method:
+            # Mock breaks the internals of the `score` method
+            try:
+                trained_model.score(DUMMY_TS_DATAFRAME)
+            except AssertionError:
+                pass
+
+            (call_df,) = patch_method.call_args[0]
+
+            for j in DUMMY_TS_DATAFRAME.item_ids:
+                truncated_data = DUMMY_TS_DATAFRAME.loc[j][: -trained_model.prediction_length]
+                assert np.allclose(call_df.loc[j], truncated_data, equal_nan=True)
+
+    def test_when_models_saved_then_they_can_be_loaded(self, trained_model):
+        trained_model.save()
+
+        loaded_model = trained_model.__class__.load(path=trained_model.path)
+
+        assert dict_equal_primitive(trained_model.params, loaded_model.params)
+        assert dict_equal_primitive(trained_model.params_aux, loaded_model.params_aux)
+        assert trained_model.metadata == loaded_model.metadata
+        for orig_oof_pred, loaded_oof_pred in zip(
+            trained_model.get_oof_predictions(), loaded_model.get_oof_predictions()
+        ):
+            assert orig_oof_pred.equals(loaded_oof_pred)
+
+    @pytest.mark.parametrize(
+        "test_data",
+        [
+            get_data_frame_with_item_index(["A", "B"], data_length=15),
+            mask_entries(get_data_frame_with_item_index(["C", "D"], data_length=60)),
+            get_data_frame_with_item_index(["A"], data_length=10),
+            get_data_frame_with_item_index([0, 1, 2, 3], data_length=15),
+        ],
+    )
+    def test_when_predict_called_then_predictor_inference_aligns_index_aligns_with_expected_index(
```

Comment:
Not sure if I understand the meaning of `predictor_inference` here. Maybe `test_when_predict_called_then_forecast_index_aligns_with_expected_index`?